homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -43,19 +45,35 @@ class SpeedtestSensor(CoordinatorEntity, RestoreEntity, SensorEntity):
     """Implementation of a speedtest.net sensor."""
 
     coordinator: SpeedTestDataCoordinator
+    entity_description: SpeedtestSensorEntityDescription
     _attr_icon = ICON
 
     def __init__(
         self,
         coordinator: SpeedTestDataCoordinator,
-        description: SensorEntityDescription,
+        description: SpeedtestSensorEntityDescription,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_name = f"{DEFAULT_NAME} {description.name}"
         self._attr_unique_id = description.key
         self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, "Speed Test")},
+            "name": DEFAULT_NAME,
+            "entry_type": "service",
+        }
+
+    @property
+    def native_value(self) -> StateType:
+        """Return native value for entity."""
+        if self.coordinator.data:
+            state = self.coordinator.data[self.entity_description.key]
+            self._attr_native_value = cast(
```

Comment:
Why do we set the attribute in the property function?
I think we should either not implement a `native_value` property function and just set the `_attr_native_value` when there are updates, or the opposite, but not mix them to make the code easier to follow.

If we want the `native_value` property function, consider renaming the `_attr_native_value`.